src/Umbraco.Web.Website/Extensions/WebsiteUmbracoBuilderExtensions.cs

```diff
@@ -11,12 +12,12 @@ namespace Umbraco.Extensions
     /// <summary>
     /// <see cref="IUmbracoBuilder"/> extensions for umbraco front-end website
     /// </summary>
-    public static class UmbracoBuilderExtensions
+    public static class WebsiteUmbracoBuilderExtensions
```

Comment:
This should live with other DI code.

Reply:
Sounds good - how do you recommend we organize this best then? 

Update: ah, I see your comment here https://github.com/umbraco/Umbraco-CMS/pull/9589/files#r547360727 = `Umbraco.Web.DependencyInjection` 

With the snapshot of code in this PR we have:

* Project Umbraco.Web.Common - `Umbraco.Web.Common.Extensions.UmbracoBuilderExtensions`
* Project Umbraco.Web.Website - `Umbraco.Extensions.WebsiteUmbracoBuilderExtensions`
* Project Umbraco.Web.BackOffice- `Umbraco.Extensions.BackOfficeUmbracoBuilderExtensions`

Just looking at where other DI code exists in these projects and its still in various places while we are cleaning things up. Just noticed we still have `Umbraco.Web.Common.Runtime.AspNetCoreComposer` which can easily be removed and transitioned to a builder extension. 

So should the end result be this?

* Project Umbraco.Web.Common - `Umbraco.Web.DependencyInjection.UmbracoBuilderExtensions` - `UmbracoBuilderExtensions_Common.cs`
* Project Umbraco.Web.Website - `Umbraco.Web.DependencyInjection.UmbracoBuilderExtensions` - `UmbracoBuilderExtensions_Website.cs`
* Project Umbraco.Web.BackOffice- `Umbraco.Web.DependencyInjection.UmbracoBuilderExtensions` - `UmbracoBuilderExtensions_BackOffice.cs`

Reply:
Looking at the file contents there is actually a lot less code than I would have anticipated. When I suggested the partial classes I thought there would be a lot more.

Normally I would say namespace to match your binary and location but the namespaces of those three projects are all over the place - and, because the name namespace is spread across the projects you cannot use a single partial. 

I won't lie.... I do not understand the hierarchy.

- `Umbraco.Web.BackOffice` references `Umbraco.Web.Common`, and, `Umbraco.Examine.Lucene`
- `Umbraco.Web.Common` references `Umbraco.Core` and `Umbraco.Infrastructure`
- `Umbraco.Web` references `Umbraco.Core` , `Umbraco.Infrastructure`, and `Umbraco.Persistance.SqlCe`

This means that there is nothing common about the `Umbraco.Web.Common` project since it is not share by both the `Umbraco.Web.xxx` projects!

Personally I would fix up the project names and namespaces (_since they're not even consistent within the projects own classes_) and do this:

`Umbraco.{NAMESPACE}.DependencyInjection.UmbracoBuilderExtensions` with the standard filename.

Reply:
Yep namespaces are in chaos currently. There's some decisions to be made about the organization and then a big cleanup. This has been avoided currently out of merging pains but will need to happen sooner than later for sure, driving me (and probably everyone else) mad! 

 `Umbraco.Web` - is the old net framework project, it will die

The front-end one is `Umbraco.Web.Website` which does reference common.

The reason that `Umbraco.Extensions` exists between projects as a global namespace is to make `using` requirements for end users very simple.

For now though I will use the `Umbraco.{NAMESPACE}.DependencyInjection.UmbracoBuilderExtensions` for UmbracoBuilder extensions and shuffle a bit more code to align with that namespace.

Reply:
>The reason that Umbraco.Extensions exists between projects as a global namespace is to make using requirements for end users very simple.

With good namespace hierarchy and API docs you can avoid this. 

The namespace fixup has to come soon. It makes design and discovery so much harder currently!

Reply:
Namespace pain put a stop to #9396

It was hinted that both Umbraco.Core and Umbraco.Infrastructure will both be set to have a shared namespace of Umbraco.Cms which kindof confuses me, any ideas what that's about?

Reply:
The plan is to make the base namespace Umbraco.Cms along with the nuget package name then streamline the other umbraco products like UmbracoDeploy to be Umbraco.Deploy namespace + package name, same goes with UmbracoForms to Umbraco.Forms. 

Not 100% sure about the Core vs Infra projects sharing the base namespace. I presume the concept is that Umbraco.Core and Umbraco.Infrastructure projects are sort of 'the same' and there shouldn't be an 'Infrastructure' namespace but these projects are separated due to dependency requirements and hopefully over time more would be moved to the Core project where dependencies are not needed (i.e. Services). We should have a chat about what the intention is early in the NY with @bergmania and the netcore team

Reply:
The original idea with shared namespaces was back from the first RFC. 
Because, as @Shazwazza says, the concept is sort of the same and we don't expect to have everything that should be in Core, located in core in the first release. This would make breaking changes easier to resolve.

Reply:
I can understand there's a lot in Infra namespace that shouldn't be there, but not so much that infra namespace should exist.

FileSystemMediaService / AzureBlobStorageMediaService / S3MediaService would all be examples I'd expect to find in a infra namespace

Reply:
The lack of decent namespace organisation is honestly intolerable. It’s hurting discovery and development. 

This weird development-by-part approach is quickly turning into a demonstration of the sunken cost fallacy. The codebase is an utter mess! Accept that and fix it properly rather than shipping a have baked result.

Reply:
lol @JimBobSquarePants you're so often right, but there's that catch more with honey than vinegar thing